src/Microsoft.AspNetCore.Razor.Runtime/Hosting/RazorSourceThumbprintAttribute.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Hosting
+{
+    /// <summary>
+    /// Specifies the hash thumprint of a source file that contributed to a compiled item.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// These attributes are added by the Razor infrastructure when generating code to assist runtime
+    /// implementations to determine the integrity of compiled items.
+    /// </para>
+    /// <para>
+    /// Runtime implementations should access the thumbprint metadata for an item using
+    /// <see cref="RazorCompiledItemExtensions.GetThumprintMetadata(RazorCompiledItem)"/>.
+    /// </para>
+    /// </remarks>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = true)]
```

Comment:
I was torn on whether or not to include other type-like things here other than class, so YAGNI kicked in. If someone wants to make an argument, I can be convinced.

Reply:
Other things would be enum, interface, struct, etc.